src/core/idb-cache.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * A promise based interface to a key-value based indexedDB cache
+ */
+
+const databases = new Map();
+
+export default class IDBCache {
+  /**
+   * @constructor
+   * @param {string} name             name of database
+   * @param {array} stores            array of names of stores
+   * @param {object} Options
+   *    @param {integer} version      database version
+   *    @param {string} defaultStore  name of default store
+   */
+  constructor(
+    name,
+    stores,
+    { version, defaultStore } = { version: 1, defaultStore: null }
+  ) {
+    this.name = name;
+    if (!Array.isArray(stores) || stores.length < 1) {
+      throw new TypeError("`stores` must be an array of length atleast 1");
+    }
+    this.stores = [...stores];
+    this.version = version;
+    this.defaultStore = defaultStore || this.stores[0];
```

Comment:
Maybe here we should just do the following, so `.ready` can just be a property:

```KS
this.ready = new Promise(async resolve=>{
  await databases.get(name);
  resolve();
});
```

Eventually, we should do this properly with a private field and getter. We would need to add the right Babel plugin.

Reply:
updated the above